app/scripts/modules/core/src/pipeline/config/triggers/cron/CronAdvance.tsx

```diff
@@ -0,0 +1,126 @@
+import * as React from 'react';
+
+import { Observable, Subject } from 'rxjs';
+import { capitalize } from 'lodash';
+
+import { HelpField } from 'core/help';
+import { CronValidatorService } from './cronValidator.service';
+import { ICronTriggerConfigProps } from './cronConfig';
+
+export interface ICronAdvanceState {
+  description?: string;
+  errorMessage?: string;
+}
+
+export class CronAdvance extends React.Component<ICronTriggerConfigProps, ICronAdvanceState> {
+  private destroy$ = new Subject();
+
+  constructor(props: ICronTriggerConfigProps) {
+    super(props);
+    this.state = {
+      description: '',
+      errorMessage: '',
+    };
+  }
+
+  public componentDidMount(): void {
+    this.validateCronExpression(this.props.trigger.cronExpression);
+  }
+
+  public componentWillUnmount(): void {
+    this.destroy$.next();
+  }
+
+  private validateCronExpression = (cronExpression: string) => {
+    Observable.fromPromise(CronValidatorService.validate(cronExpression))
+      .takeUntil(this.destroy$)
+      .subscribe(
+        (result: { valid: boolean; message?: string; description?: string }) => {
+          if (result.valid) {
+            this.setState({
+              description: result.description ? capitalize(result.description) : '',
+              errorMessage: '',
+            });
+          } else {
+            this.setState({
+              description: '',
+              errorMessage: result && result.message ? result.message : 'Error validating CRON expression',
+            });
+          }
+        },
+        (result: { valid: boolean; message?: string; description?: string }) => {
+          this.setState({
+            description: '',
+            errorMessage: result && result.message ? result.message : 'Error validating CRON expression',
+          });
+        },
+      );
+  };
+
+  private onUpdateTrigger = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const cronExpression = event.target.value.replace(/\s\s+/g, ' ');
+    this.props.triggerUpdated({
+      ...this.props.trigger,
+      cronExpression,
+    });
+    this.validateCronExpression(cronExpression);
+  };
+
+  public render() {
+    const { cronExpression } = this.props.trigger;
+    const { description, errorMessage } = this.state;
+    return (
+      <div>
+        <div className="row">
+          <div className="col-md-12">
+            <strong>Expression</strong>
+            <HelpField id="pipeline.config.cron.expression" />
+            &nbsp;
```

Comment:
minor nit (and not documented except in code): we've been trying to standardize on `{' '}` rather than `&nbsp;` to force white space between elements.

Reply:
👍